pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/FileUploadPathProvider.java

```diff
@@ -66,22 +67,32 @@ public FileUploadPathProvider(ControllerConf controllerConf) throws InvalidContr
       Utils.rethrowException(e);
     }
 
+    String localTempDir = _controllerConf.getLocalTempDir();
+    if (localTempDir == null) {
+      LOGGER.warn("Local temp directory not set, setting to data dir {}", dataDir);
+      localTempDir = dataDir;
+    }
+
     PinotFS pinotFS = PinotFSFactory.create(scheme);
 
     try {
+      _localTempDirURI = new URI(localTempDir);
+      if (!pinotFS.exists(_localTempDirURI)) {
+        pinotFS.mkdir(_localTempDirURI);
+      }
       _baseDataDirURI = new URI(dataDir);
       if (!pinotFS.exists(_baseDataDirURI)) {
         pinotFS.mkdir(_baseDataDirURI);
```

Comment:
As a side note, I see that mkdir() will forcefully remove the directory if it already exists. This seems quite dangerous to me. For example, if there's a bug in pinotFs.exists(), we will end up deleting the data dir. This might be acceptable behavior for a file, but dangerous for deleting the entire directory. Let's open an issue to visit this outside of this PR.